core/src/main/java/lucee/runtime/functions/string/StringReduce.java

```diff
@@ -0,0 +1,31 @@
+package lucee.runtime.functions.string;
+
+import lucee.runtime.PageContext;
+import lucee.runtime.exp.FunctionException;
+import lucee.runtime.exp.PageException;
+import lucee.runtime.ext.function.BIF;
+import lucee.runtime.functions.closure.Reduce;
+import lucee.runtime.op.Caster;
+import lucee.runtime.type.UDF;
+import lucee.runtime.type.util.StringListData;
+
+public class StringReduce extends BIF {
+	private static final long serialVersionUID = -2153555241217815037L;
+
+	@Override
+	public Object invoke(PageContext pc, Object[] args) throws PageException {
+		if (args.length < 2) {
+			throw new FunctionException(pc, "StringSome", 2, 3, args.length);
+		}
+		return call(pc, Caster.toString(args[0]), Caster.toFunction(args[1]), args[2]);
```

Comment:
that will fail when we only have 2 arguments in the request, because line 3 is expecting 3 arguments and line 17 only handles less that 2.
i mostly do it the other way arround, so
if(args.length==2) return call(pc, Caster.toString(args[0]), Caster.toFunction(args[1]));
else if(args.length==3) return call(pc, Caster.toString(args[0]), Caster.toFunction(args[1]), args[2]);
else throw new FunctionException(pc, "StringSome", 2, 3, args.length);

Reply:
correction:
if(args.length==2) return call(pc, Caster.toString(args[0]), Caster.toFunction(args[1]),null);
else if(args.length==3) return call(pc, Caster.toString(args[0]), Caster.toFunction(args[1]), args[2]);
else throw new FunctionException(pc, "StringSome", 2, 3, args.length);